docs/source/reference/api-server/api-server-tunning.rst

```diff
@@ -0,0 +1,152 @@
+.. _sky-api-server-performance-best-practices:
+
+SkyPilot API Server Performance Best Practices
+==============================================
+
+This page describes performance best practices for remote SkyPilot API server in :ref:`team deployment <sky-api-server>`.
+
+Tuning API server resources
+---------------------------
+
+The number of requests that the API server can handle concurrently is proportional to the resources (CPU cores and memory) allocated to it. Internally, requests are categorized into two different types and handled in a first-in-first-out manner:
+
+* ``Long-running request``: request that takes long time and more resources to run, including ``launch``, ``exec``, ``jobs.launch``, etc.
+* ``Short-running request``: request that takes short time or less resources to run, including ``status``, ``logs``, etc.
+
+.. note::
+
+    Though a task (or job) can run for any length of time, concurrent tasks does not occupy the concurrency. Because once a task is submitted to the cluster, it will be detached and no longer takes any resources off the API server.
+
+It is recommended to tune the resources allocated to the API server based on the expected concurrency to avoid queuing:
+
+.. tab-set::
+
+    .. tab-item:: Helm Deployment
+
+        .. code-block:: bash
+
+            # NAMESPACE and RELEASE_NAME should be the same as the deployment step
+            helm upgrade -n ${NAMESPACE} ${RELEASE_NAME} skypilot/skypilot-nightly \
+                --reuse-values \
+                --set apiService.resources.requests.cpu=8 \
+                --set apiService.resources.requests.memory=16Gi \
+                --set apiService.resources.limits.cpu=8 \
+                --set apiService.resources.limits.memory=16Gi
+            
+        .. note:: 
+
+            If you specify a resources that is lower than the minimum recommended resources (4 CPUs with 8GB of memory) for team usage, an error will be raised on ``helm upgrade``. You can specify ``--set apiService.skipResourcesCheck=true`` to skip the check if performance and stability is not an issue for you scenario.
```

Comment:
It's a bit confusing that " (4 CPUs with 8GB of memory)" is mentioned as our rec settings, but snippet above doesn't reflect this?

We should mention the rec setting in a 1-sentence paragraph.

Reply:
Sounds good, please go ahead.